dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/apps/view/AppsInterpolationTest.java

```diff
@@ -0,0 +1,112 @@
+package com.dotcms.rest.api.v1.apps.view;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import com.dotcms.datagen.AppDescriptorDataGen;
+import com.dotcms.rest.api.v1.DotObjectMapperProvider;
+import com.dotcms.security.apps.AppDescriptor;
+import com.dotcms.security.apps.ParamDescriptor;
+import com.dotcms.security.apps.Secret;
+import com.dotcms.security.apps.Type;
+import com.dotcms.util.IntegrationTestInitService;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class AppsInterpolationTest {
+
+
+    @BeforeClass
+    public static void prepare() throws Exception {
+        //Setting web app environment
+        IntegrationTestInitService.getInstance().init();
+    }
+
+    static final ObjectMapper mapper = DotObjectMapperProvider.getInstance()
+            .getDefaultObjectMapper();
+
+    /**
+     * Given scenario: we build an AppView json based on an AppDescriptor. This fires the serializers so that any embedded velocity code will br processed
+     * Expected Results: No velocity variable declared in the data model used to create the initial json such as `$myva` should make it in the result.
+     * Also we validate the replacements individually.
+     * @throws JsonProcessingException
+     */
+    @Test
+    public void Test_Interpolation() throws JsonProcessingException {
+
+        final List<Map<String, String>> list = ImmutableList.of(
+                ImmutableMap.of("label", "-", "value", ""),
```

Comment:
Codacy found an issue: [The String literal "label" appears 6 times in this file; the first occurrence is on line 47](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6392890)